lib/middleware/fireAlarmMiddleware.ts

```diff
@@ -0,0 +1,192 @@
+    targetVersionString.indexOf('=') + 1
+  );
+
+  const targetVersionParts = targetVersion.split('.');
+  const versionParts = version.split('.');
+
+  // Require the wildcard to be explicitly set to target all versions
+  if (
+    versionParts.length < 3 &&
+    versionParts[versionParts.length - 1] !== WILDCARD
+  ) {
+    return false;
+  }
+
+  // Don't allow the less than or equal to operator to be used with the wildcard
+  // in the major version part (e.g. <=*)
+  if (targetVersionString.startsWith('<=') && versionParts[0] === WILDCARD) {
+    return false;
+  }
+
+  let targetAnyVersion = false;
+
+  return targetVersionParts.every((targetPart, i) => {
+    const versionPart = versionParts[i];
+
+    // Support generic version targeting (like 1.2.*)
+    if (targetPart === WILDCARD || targetAnyVersion) {
+      targetAnyVersion = true;
+      return true;
+    }
+
+    // Double check that the target part is a number
+    if (isNaN(Number(targetPart))) {
+      return false;
+    }
+
+    return targetVersionString.startsWith('<=')
+      ? Number(versionPart) <= Number(targetPart)
+      : Number(versionPart) === Number(targetPart);
+  });
+}
+
+function isCommandTargeted(
+  command: string,
+  targetCommandsString: string | null
+): boolean {
+  // Require the wildcard to be explicitly set to target all commands
+  if (!targetCommandsString) {
+    return false;
+  }
+
+  if (targetCommandsString === WILDCARD) {
+    return true;
+  }
+
+  const targetCommands = targetCommandsString.split(',');
+
+  return targetCommands.some(cmd => command.startsWith(cmd));
+}
+
+function filterFireAlarm(
+  fireAlarm: FireAlarm,
+  command: string,
+  version: string
+): boolean {
+  const commandIsTargeted = isCommandTargeted(command, fireAlarm.querySelector);
+  const versionIsTargeted = isVersionTargeted(
+    version,
+    fireAlarm.urlRegexPattern
+  );
+
+  return commandIsTargeted && versionIsTargeted;
+}
+
+async function getFireAlarms(
+  accountId: number,
+  command: string,
+  version: string
+): Promise<FireAlarm[]> {
+  let relevantAlarms: FireAlarm[] = [];
+
+  try {
+    const { data: fireAlarms } = await fetchFireAlarms(accountId);
+
+    relevantAlarms = fireAlarms.filter(fireAlarm =>
+      filterFireAlarm(fireAlarm, command, version)
+    );
+  } catch (error) {
+    debugError(error);
+  }
+
+  return relevantAlarms;
+}
+
+async function logFireAlarms(
+  accountId: number,
+  command: string,
+  version: string
+): Promise<void> {
+  const alarms = await getFireAlarms(accountId, command, version);
+
+  if (alarms.length > 0) {
+    const notifications = alarms.reduce((acc, alarm) => {
+      if (alarm.title && alarm.message) {
+        return (
+          acc +
+          `${acc.length > 0 ? '\n\n' : ''}${chalk.bold(alarm.title)}\n${alarm.message}`
+        );
+      }
+      return acc;
+    }, '');
+
+    let boxen;
+
+    try {
+      boxen = (await import('boxen')).default;
```

Comment:
OOC why are we dynamically loading the library?

Reply:
That package has a type of "module" which wasn't playing nicely with our "nodeNext". Here's the error:


> The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("boxen")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to 'package.json'.ts(1479)

I'm pretty sure we don't want to change our package type to module because then we'd need to include file extensions in our imports. It creates a whole mess of other problems.

Reply:
Gotcha, thanks for explaining